pkg/planner/cardinality/row_count_column.go

```diff
@@ -171,14 +171,20 @@ func equalRowCountOnColumn(sctx planctx.PlanContext, c *statistics.Column, val t
 		return histCnt, nil
 	}
 	// 3. use uniform distribution assumption for the rest (even when this value is not covered by the range of stats)
+	// branch1: histDNV <= 0 means that all NDV's are in TopN, and no histograms.
+	// branch2: histDNA > 0 basically means while there is still a case, c.Histogram.NDV >
+	// c.TopN.Num() a little bit, but the histogram is still empty. In this case, we should use the branch1 and for the diff
+	// in NDV, it's mainly comes from the NDV is conducted and calculated ahead of sampling.
 	histNDV := float64(c.Histogram.NDV - int64(c.TopN.Num()))
-	if histNDV <= 0 {
+	if histNDV <= 0 || (c.IsFullLoad() && c.Histogram.NotNullCount() == 0) {
```

Comment:
Can we add some test cases for it?

Reply:
Manual test is shown above, need a local stats.json